test/hotspot/jtreg/compiler/loopopts/superword/TestUnorderedReduction.java

```diff
@@ -38,9 +62,22 @@ public class TestUnorderedReduction {
     static final int ITER  = 10;
 
     public static void main(String[] args) {
-        TestFramework.runWithFlags("-Xbatch",
-                                   "-XX:CompileCommand=compileonly,compiler.loopopts.superword.TestUnorderedReduction::test*",
-                                   "-XX:MaxVectorSize=16");
+        TestFramework framework = new TestFramework(TestUnorderedReduction.class);
+        framework.addFlags("-Xbatch",
+                           "-XX:CompileCommand=compileonly,compiler.loopopts.superword.TestUnorderedReduction::test*");
+
+        if (args.length != 1) {
+            throw new RuntimeException("Test requires exactly one argument!");
+        }
+
+        switch (args[0]) {
+            case "Vanilla-Unaligned"         -> { framework.addFlags("-XX:-AlignVector"); }
+            case "Vanilla-Aligned"           -> { framework.addFlags("-XX:+AlignVector"); }
+            case "MaxVectorSize16-Unaligned" -> { framework.addFlags("-XX:-AlignVector", "-XX:MaxVectorSize=16"); }
+            case "MaxVectorSize32-Aligned"   -> { framework.addFlags("-XX:+AlignVector", "-XX:MaxVectorSize=32"); }
```

Comment:
Since `AlignVector` and `MaxVectorSize` are C2 flags, you should add a `@requires vm.compiler2.enabled` to all `@test` definitions.

Reply:
good idea, done.